pre_commit/commands/autoupdate.py

```diff
@@ -84,7 +84,9 @@ def _check_hooks_still_exist_at_rev(
         )
 
 
-REV_LINE_RE = re.compile(r'^(\s+)rev:(\s*)([^\s#]+)(.*)(\r?\n)$', re.DOTALL)
+REV_LINE_RE = re.compile(
+    r'^(\s+)rev:(\s*)([\'"]?)([^\s#]+)([\'"]?)(.*)(\r?\n)$', re.DOTALL,
```

Comment:
git tags can contain and end with quotes so this regex isn't quite correct

Reply:
Ah, definitely hadn't considered that. On the basis that leading quotes in the tag would need to be escaped and so correctly not match, and that there's no benefit to trying (and currently failing due to greedy regex) to match any closing quotes, did you have this in mind? (dropping the second quote match)
```
^(\s+)rev:(\s*)([\'"]?)([^\s#]+)(.*)(\r?\n)$
```

Reply:
yeah I think that would work, I can't think of any problems with that off the top of my head 🤔